server/php-worker.cpp

```diff
@@ -154,12 +157,13 @@ void PhpWorker::state_init_script() noexcept {
 
   script_t *script = get_script();
   dl_assert(script != nullptr, "failed to get script");
-  if (php_script == nullptr) {
-    php_script = new PhpScript(max_memory, oom_handling_memory_ratio, 8 << 20);
+  if (!php_script.has_value()) {
+    php_script.emplace(max_memory, oom_handling_memory_ratio, 8 << 20);
   }
   dl::init_critical_section();
-  php_script->init(script, data);
-  php_script->set_timeout(timeout);
+  php_script.value().init(script, data);
```

Comment:
`std::optional<>::value()` checks that it has value at first. But here we can guarantee it because of the condition above, so let's use just [`operator ->`](https://en.cppreference.com/w/cpp/utility/optional/operator*) here and at other such places. As for me, It will look much better)

Reply:
btw, you forgot to `std::move(data)` here, but anyway we discussed another aproach